react-components/src/hooks/useSearchMappedEquipmentFDM.tsx

```diff
@@ -101,16 +109,24 @@ export const useAllMappedEquipmentFDM = (
 
       let currentPage = await fdmSdk.queryNodesAndEdges(query);
 
-      const mappedEquipment = currentPage.items.mapped_nodes.map((node) =>
-        removeEmptyProperties(node)
-      );
+      const mappedNodes = currentPage.items.mapped_nodes as NodeItem[];
+      const mappedNodesParents = currentPage.items.mapped_nodes_2 as NodeItem[];
+      const mappedEquipment = mappedNodes
+        .concat(mappedNodesParents)
+        .map((node) => removeEmptyProperties(node));
 
       while (!isEqual(currentPage.nextCursor, {})) {
         query.cursors = currentPage.nextCursor;
 
         currentPage = await fdmSdk.queryNodesAndEdges(query);
 
-        for (const node of currentPage.items.mapped_nodes) {
+        for (const node of currentPage.items.mapped_nodes as NodeItem[]) {
+          const cleanedNode = removeEmptyProperties(node);
+
+          mappedEquipment.push(cleanedNode);
+        }
+
+        for (const node of currentPage.items.mapped_nodes_2 as NodeItem[]) {
           const cleanedNode = removeEmptyProperties(node);
```

Comment:
I can, just thought that it would be be more performant to do it like this since destructuring and pushing to the array like this is not very efficient as far as I know.

Reply:
Changed to that 
```
 const cleanedNodes = currentPage.items.mapped_nodes.map((node) => removeEmptyProperties(node as NodeItem));
 const cleanedNodesParents = currentPage.items.mapped_nodes_2.map((node) => removeEmptyProperties(node as NodeItem));

 mappedEquipment.push(...cleanedNodes.concat(cleanedNodesParents));
```

Reply:
Just looked it up - `concat` actually creates a new array, it does not append to another.. :sweat_smile: I still think we should use `push` only once, since it may allow the array implementation to not re-allocate if the number of elements grows out of bounds multiple times. Can we do `mappedEquipment.push(...cleanedNodes, ...cleanedNodesParents)`? May still be slow, but at least it's one less allocation.. Probably :thinking:

Reply:
I think it should be fine as is, will change it in the future PR probably.